meta/sai_rpc_frontend.cpp

```diff
@@ -0,0 +1,870 @@
+      free(sai_attr.value.aclresource.list);
+    } break;
+    case SAI_ATTR_VALUE_TYPE_IP_ADDRESS_LIST: {
+      for (unsigned int i = 0; i < sai_attr.value.ipaddrlist.count; i++) {
+        sai_thrift_ip_address_t thrift_ip;
+        sai_ip_address_t_to_thrift(thrift_ip,
+                                   sai_attr.value.ipaddrlist.list[i]);
+        thrift_attr.value.ipaddrlist.addresslist.push_back(thrift_ip);
+      }
+      thrift_attr.value.ipaddrlist.count = sai_attr.value.ipaddrlist.count;
+      free(sai_attr.value.ipaddrlist.list);
+    } break;
+    case SAI_ATTR_VALUE_TYPE_MAP_LIST:
+    case SAI_ATTR_VALUE_TYPE_VLAN_LIST:
+    case SAI_ATTR_VALUE_TYPE_QOS_MAP_LIST: {
+      for (unsigned int i = 0; i < sai_attr.value.qosmap.count; i++) {
+        sai_thrift_qos_map_t thrift_qos_map;
+        // key
+        thrift_qos_map.key.tc = sai_attr.value.qosmap.list[i].key.tc;
+        thrift_qos_map.key.dscp = sai_attr.value.qosmap.list[i].key.dscp;
+        thrift_qos_map.key.dot1p = sai_attr.value.qosmap.list[i].key.dot1p;
+        thrift_qos_map.key.prio = sai_attr.value.qosmap.list[i].key.prio;
+        thrift_qos_map.key.pg = sai_attr.value.qosmap.list[i].key.pg;
+        thrift_qos_map.key.queue_index =
+            sai_attr.value.qosmap.list[i].key.queue_index;
+        thrift_qos_map.key.color =
+            static_cast<int32_t>(sai_attr.value.qosmap.list[i].key.color);
+        thrift_qos_map.key.mpls_exp =
+            sai_attr.value.qosmap.list[i].key.mpls_exp;
+        // value
+        thrift_qos_map.value.tc = sai_attr.value.qosmap.list[i].value.tc;
+        thrift_qos_map.value.dscp = sai_attr.value.qosmap.list[i].value.dscp;
+        thrift_qos_map.value.dot1p = sai_attr.value.qosmap.list[i].value.dot1p;
+        thrift_qos_map.value.prio = sai_attr.value.qosmap.list[i].value.prio;
+        thrift_qos_map.value.pg = sai_attr.value.qosmap.list[i].value.pg;
+        thrift_qos_map.value.queue_index =
+            sai_attr.value.qosmap.list[i].value.queue_index;
+        thrift_qos_map.value.color =
+            static_cast<int32_t>(sai_attr.value.qosmap.list[i].value.color);
+        thrift_qos_map.value.mpls_exp =
+            sai_attr.value.qosmap.list[i].value.mpls_exp;
+        thrift_attr.value.qosmap.maplist.push_back(thrift_qos_map);
+      }
+      thrift_attr.value.qosmap.count = sai_attr.value.qosmap.count;
+      free(sai_attr.value.qosmap.list);
+    } break;
+    case SAI_ATTR_VALUE_TYPE_TLV_LIST:
+    case SAI_ATTR_VALUE_TYPE_SEGMENT_LIST:
+    case SAI_ATTR_VALUE_TYPE_PORT_EYE_VALUES_LIST:
+    case SAI_ATTR_VALUE_TYPE_TIMESPEC:
+    case SAI_ATTR_VALUE_TYPE_NAT_ENTRY_DATA:
+      // not supported
+      break;
+    default:
+      break;
+  }
+}
+
+// including it here we never have to modify the generated file
+#include "sai_rpc_server.cpp"
+
+class sai_rpcHandlerFrontend : virtual public sai_rpcHandler {
+  int64_t sai_thrift_object_type_get_availability(
```

Comment:
curious why this (and others) had to be written manually.

Reply:
These two manually written 'sai_thrift_' functions are specific and they don't match the patterns the generator is working with. There are only two of such functions related to CRM verification.